src/aosm/azext_aosm/cli_handlers/onboarding_vnf_handler.py

```diff
@@ -60,104 +34,7 @@ def output_folder_file_name(self) -> str:
-            )
-
-        # Instantiate vhd processor
-        if not self.config.vhd.artifact_name:
-            self.config.vhd.artifact_name = self.config.nf_name + "-vhd"
-        vhd_processor = VHDProcessor(
-            name=self.config.vhd.artifact_name,
-            input_artifact=VHDFileInput(
-                artifact_name=self.config.vhd.artifact_name,
-                artifact_version=self.config.vhd.version,
-                default_config=self._get_default_config(self.config.vhd),
-                file_path=Path(self.config.vhd.file_path).absolute(),
-                blob_sas_uri=self.config.vhd.blob_sas_url,
-            ),
-        )
-        processor_list.append(vhd_processor)
-        return processor_list
-
-    def build_base_bicep(self):
-        """Build the base bicep file."""
-        # Build manifest bicep contents, with j2 template
-        template_path = self._get_template_path(
-            VNF_TEMPLATE_FOLDER_NAME, VNF_BASE_TEMPLATE_FILENAME
-        )
-        bicep_contents = self._render_base_bicep_contents(template_path)
-        # Create Bicep element with manifest contents
-        bicep_file = BicepDefinitionElementBuilder(
-            Path(VNF_OUTPUT_FOLDER_FILENAME, BASE_FOLDER_NAME), bicep_contents
-        )
-        return bicep_file
-
-    def build_manifest_bicep(self):
-        """Build the manifest bicep file."""
-        acr_artifact_list = []
-
-        logger.info("Creating artifact manifest bicep")
-
-        for processor in self.processors:
-            if isinstance(processor, BaseArmBuildProcessor):
-                acr_artifact_list.extend(processor.get_artifact_manifest_list())
-                logger.debug(
-                    "Created list of artifacts from %s arm template(s) provided: %s",
-                    len(self.config.arm_templates),
-                    acr_artifact_list,
-                )
-            elif isinstance(processor, VHDProcessor):
-                sa_artifact_list = processor.get_artifact_manifest_list()
-                logger.debug(
-                    "Created list of artifacts from vhd image provided: %s",
-                    sa_artifact_list,
-                )
-
-        # Build manifest bicep contents, with j2 template
-        template_path = self._get_template_path(
-            VNF_TEMPLATE_FOLDER_NAME, VNF_MANIFEST_TEMPLATE_FILENAME
-        )
-        bicep_contents = self._render_manifest_bicep_contents(
-            template_path, acr_artifact_list, sa_artifact_list
-        )
-        # Create Bicep element with manifest contents
-        bicep_file = BicepDefinitionElementBuilder(
-            Path(VNF_OUTPUT_FOLDER_FILENAME, MANIFEST_FOLDER_NAME),
-            bicep_contents,
-        )
-
-        logger.info("Created artifact manifest bicep element")
-        return bicep_file
-
-    def build_artifact_list(self):
+    def build_artifact_list(self) -> ArtifactDefinitionElementBuilder:
         """Build the artifact list."""

```

Comment:
Could we get some better doc strings? First line is fine, but could you add another paragraph saying where/what the artifact list is used for?